### ISSUE TITLE
Allow Solder Temp & Boost Temp an option to type exact number directly into the box for changing temperature

### ISSUE THREAD

**Issue by River-b** [builder555/PineSAM#12]
sorry for submitting all of these at once. I just had some time today to do more testing with the app.

**Is your feature request related to a problem? Please describe.**
Sometimes it's hard for my mouse hand to easily slide the bar to change the temperature setting here accurately, i.e. I want to set it to 306 C. I suspect this would also be same if I got this working on Android with touch screen too.

**Describe the solution you'd like**
 Is it possible for the number boxes on Soldering temp and Boost Temp to allow people choice to type the exact number desired into the box instead of using the slider (both number entry or slider as a choice)?

![image](https://user-images.githubusercontent.com/97197236/219804062-7040749b-f6a2-4416-9c95-0562a5982373.png)

**Additional context**
I still like slider too, just sometimes mouse is frustrating and would be good to just type it in. I think on Android phone I would more likely use the slider but on PC with keyboard, typing a number directly into the box is faster/more accurate.

**Comment by builder555** [comment]
no worries. it'll give me something to work on during the weekend :)

**Comment by builder555** [comment]
You should be able to click the value and change it directly. The value will be saved once you navigate out of the box.
the commit automatically closed the issue, but feel free to re-open it if something else is missing @River-b

**Comment by River-b** [comment]
new download does not seem to be working for both my Pinecils.
Both pinecils work on Joric's BLE. Both have same beta  dated feb 18, 2023 (more unique).

One of my Pinecils works on PineSam ( I downloaded fresh zip from main page 20 min ago to test temperature change feature).

Second V2, does not and gets these errors
![image](https://user-images.githubusercontent.com/97197236/219975458-9f46ea65-eded-4d31-97b7-f1147b5d7f97.png)
![image](https://user-images.githubusercontent.com/97197236/219975480-111c4c3d-cdab-4fdd-8287-478e42cc7d14.png)

**Comment by River-b** [comment]
nvm  after doing it a few more times it worked. maybe there is a minimum amount of time we need to give Pinecils  to be on before starting the app so they have time to properly be seen by PC.

Strange thing is  the working Pinecil, I literally  just plugged it in started the script and it worked right away.  I didn't really wait or delay.

**Comment by River-b** [comment]
Testing the new temp Box.
I prefer it does not ask for a confirmation every time.
I like that the slider you just slide the bar and let go and it would just do it as soon as you let go of mouse sliding it.
With the new temperature Box  after I hit Enter  it makes me do a separate confirmation step.

![image](https://user-images.githubusercontent.com/97197236/219976426-dcb7a19f-cce7-444d-83d5-964e21f43bf3.png)
and this is what it looks like on script 
![image](https://user-images.githubusercontent.com/97197236/219976441-668cef67-eb5d-4eca-853a-a28a518e5143.png)

**Outcome desired:**

1.  have it like the slider bar where no additional confirmation is needed.  
2. user clicks on box, enters a number, then hits  `enter` . Then enter alone should be confirmation.
3. White boxes in User interface menu are easier to see. maybe make temp change boxes also white. and the size of boxes in user interface is better too. 
4. maybe change everything to left justified?  the different items are  shifted different spots so it's harder to follow them down the page.
![image](https://user-images.githubusercontent.com/97197236/219976639-00563bff-6a97-4c7a-bbe5-bd383ed8181a.png)
5. Make the slider bar just a little longer on soldering temp and boost.
6. suggestion to remove the green background and just have thick colored boxes around section to make it less busy.

**Comment by River-b** [comment]
when I enter 455C  which is out of range I get this warning but it still sets the number in the box
![image](https://user-images.githubusercontent.com/97197236/219977002-74fd02f7-ba08-4c28-85a4-51d8ae2d33fe.png)
but leaves the pinecil unchanged.   I suggest if someone enters too high a number past the limit of 450C,  give the "value out of Range"  error message but also put number in box back to  what they had and what is seen on Pinecil Screen.  
if they had 290C before, and they enter  455C,  pop up error message and put box back at 290C  since that is what is Still on the pinecil in real life since it was unchanged.  Don't put 455C in the box

**Comment by River-b** [comment]
> but feel free to re-open it if something else is missing

I don't know how to reopen the ticket I dont seem to have a button for that :) but you can see all my testing and comments

**Comment by River-b** [comment]
I just did test all over again to see what is going on. I'm using Windows 11, Chrome browser, and windows Powershell for terminal.

Good pinecil = P-a.  Bad pinecil = P-b

- P-a  plugged in, ran script . worked Right away fast and Browser window opened with no issue.

- P-b  same issue.  I did not close my browser but I close all tabs with  PineSam open.  and I can't connect 
This is what script shows on  P-b
![image](https://user-images.githubusercontent.com/97197236/219978362-d3982a15-47a6-4963-8d0a-862b7f04dea0.png)

Do you think there is something happening for people with two or more pinecils?

**Comment by River-b** [comment]
Problem is  resolved with P-B  bad pinecil. 
Rebooted whole windows 11 PC. 
Now both P-b and P-a  pinecils  connect with Script on first attempt. 

Don't know why but  reboot fixed everything.  Suggest  this is suggestion with users who report  connection issue in script or need to try 3-4 times to connect. simple reboot of whole windows PC is solution.

**Comment by builder555** [comment]
wow that's a lot of stuff :)

ok i see what's happening with the "are you sure" box - i forgot to disable the "submit on enter" on the page and "submit" makes a call to "reset all settings". whoops!
anyway, that's fixed now in #45756ce - it'll save when you hit Enter properly without pop-ups.

I'll work on resetting the value back, if there's an error.

as for two pinecils... i wouldn't be able to troubleshoot it as I only have one. theoretically it's possible to build a custom firmware for it to advertise it as a "different" device every time, but I just don't have that kind of time, anyway, if after a reboot it works, maybe we can leave it at that for now.

as for colour and UI tweaks, can you create a separate issue for that? you can just edit your comment and cut/paste it into a new one

**Comment by River-b** [comment]
yes i'll make a new ticket about colors/format

**Comment by builder555** [comment]
let me know if entering the values works as expected now or needs further tweaking

**Comment by River-b** [comment]
1. Tested out of range **Set temperature** - works and completed. typing 455  or 5  °C pops up helpful error message and leaves the Set temperature the same as the current one.
2. Test Boost Temperature - also works, can not enter anything not in the range of 250-450 °C. 
3. Tested °F also works at the adjusted F numbers.
![image](https://user-images.githubusercontent.com/97197236/220183175-d0ff1098-ed78-4cb9-83da-44f69b27c312.png)
